src/analytics/formTracker/formTracker.ts

```diff
@@ -29,38 +33,68 @@ export class FormTracker extends BaseTracker {
         continue;
       }
 
-      if (
-        (element.type === "radio" || element.type === "checkbox") &&
-        element.checked
-      ) {
-        selectedFields.push({
-          id: element.id,
-          name: element.name,
-          value: document.querySelector(`label[for="${element.id}"]`)
-            ?.textContent
-            ? document
+      if (element.type === "checkbox") {
```

Comment:
I think it's a good idea to manage checkbox case separately. But I think also we don't need a "buffer" variable (checkboxGroups).
This array is just storing the value associated to this checkbox item. With the goal to help you to concatenate of the values together. It's something you can do directly by adding or updating an item in "selectedField".
How? With exactly the same logic.
1/ Check if you have already a selectedField item with the same id, name and type
2/ If yes, just update the value by adding the element value to the existing value of this field
3/ If no, push a new item into selectedField like we use to do.
That will simplify the function.

Reply:
I have simplified the function and updated it so that if checkbox is part of a group the that new value is just concatenated to the existing value.

Reply:
Will checkboxes in the same group , have the same ID. I dont think so ?